fabric-networking-api-v1/src/testmod/java/net/fabricmc/fabric/test/networking/login/NetworkingLoginQueryTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.networking.login;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.FutureTask;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerLoginNetworkHandler;
+import net.minecraft.util.Util;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.networking.v1.ServerLoginConnectionEvents;
+import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
+import net.fabricmc.fabric.api.networking.v1.PacketSender;
+import net.fabricmc.fabric.api.networking.v1.ServerLoginNetworking;
+import net.fabricmc.fabric.test.networking.NetworkingTestmods;
+import net.fabricmc.fabric.test.networking.play.NetworkingPlayPacketTest;
+
+public final class NetworkingLoginQueryTest implements ModInitializer {
+	@Override
+	public void onInitialize() {
+		ServerLoginConnectionEvents.LOGIN_QUERY_START.register(this::onLoginStart);
+		ServerLoginConnectionEvents.LOGIN_QUERY_START.register(this::delaySimply);
+
+		// login delaying example
+		ServerLoginNetworking.registerGlobalReceiver(NetworkingPlayPacketTest.TEST_CHANNEL, (server, handler, understood, buf, synchronizer, sender) -> {
+			if (understood) {
+				FutureTask<?> future = new FutureTask<>(() -> {
+					for (int i = 0; i <= 10; i++) {
+						Thread.sleep(300);
+						NetworkingTestmods.LOGGER.info("Delayed login for number {} 300 milliseconds", i);
```

Comment:
Disable this by default, its a fine test just dont really want this all the time when using the test mods :D